awx/ui/src/components/Lookup/HostFilterLookup.js

```diff
@@ -363,6 +381,26 @@ function HostFilterLookup({
         ]}
       >
         <ModalList>
+          {isAnsibleFactsSelected && (
+            <Alert
+              variant="info"
+              title={
+                <>
+                  {t`Searching by ansible_facts require special syntax. Refer to the`}{' '}
+                  <a
+                    href={`${getDocsBaseUrl(
+                      config
+                    )}/html/userguide/inventories.html#smart-host-filter`}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                  >
+                    {t`documentation`}
+                  </a>{' '}
+                  {t`for more info.`}
+                </>
+              }
+            />
+          )}
```

Comment:
Ideally, we'd render this inside of AdvancedSearch, so the `handleIsAnsibleFactsSelected` prop wouldn't need to be drilled down through all the intermediate components. I don't think there's a way to do that and get the layout we want though, unfortunately